kernels/ZendEngine3/fcall.c

```diff
@@ -420,7 +420,6 @@ int zephir_call_user_function(zval *object_pp, zend_class_entry *obj_ce, zephir_
 	fci.params         = NULL;
 	//fci.params: Passed as separate parameter to prevent the need to convert zval ** to zval *
 	fci.no_separation  = 1;
-	fci.symbol_table   = NULL;
```

Comment:
If you follow the execution path after the line :

fci.symbol_table = null in zephir_call_user_function

the unique usage of fci->symbol_table is in extended/fcall.c, line 144.

we never go into the code of PHP in this context, so why initialize an
attribut when it's not used.

So I'm agree I can add a :

#if PHP_VERSION_ID < 70100
  ...
#endif

to be more secure.


2016-05-07 1:56 GMT+02:00 Steffen Butzer <notifications@github.com>:

> In kernels/ZendEngine3/fcall.c
> <https://github.com/phalcon/zephir/pull/1227#discussion_r62405609>:
>
> > @@ -420,7 +420,6 @@ int zephir_call_user_function(zval *object_pp, zend_class_entry *obj_ce, zephir_
> >  	fci.params         = NULL;
> >  	//fci.params: Passed as separate parameter to prevent the need to convert zval ** to zval *
> >  	fci.no_separation  = 1;
> > -	fci.symbol_table   = NULL;
>
> Just because you saw no impact it doesn't mean there won't be one.
> Removing that would be ok if there was no version using it.
>
> Even your own code example shows why it's not a good idea:
> ZEND_ASSERT(fci->symbol_table == NULL); might fail since ->symbol_table
> might be initialized to a not-null value (due to not-zeroed memory
> allocation).
>
> So we'll have to introduce ifdef's for 7.1.X here.
> (since that won't land in PHP7.0.X exactly for that
> backwards-compatibility argument above)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/phalcon/zephir/pull/1227/files/c2d15c599eeeff207834c732d7c7cd6a79d2331a#r62405609>
>